### DISCUSSION TITLE
BTNS Anti-Spam mechanisms

### DISCUSSION THREAD

**Discussion by jdogresorg** [jdogresorg/Broadcast-Token-Naming-System#5] [Ideas]
BTNS needs an anti-spam mechanism to prevent abuse... Initially `GAS` will serve this purpose.

Given the history with Counterparty and XCP being seen as a "shitcoin"... I expect there will be some resistance to using `GAS` by some community members.

BTNS should also offer a BTC only anti-spam mechanism, where the end-user pays a bit more BTC in order to perform the action, but the need to use `GAS` can be bypassed.

This will require a bit deeper of wallet integration than just custom broadcast text... Ideally should wait until I can use `custom_outputs` feature in counterparty-lib natively.... but, might have to just roll out custom implementation like I did for the Automatic Donation System (ADS) where I just modify the unsigned tx to tack on an additional output and adjust the change value down.

Unsure if anti-spam fee should go to miners, go to dev fund, or be burned (evil laugh) ... Prolly best to just let ppl choose 

Anti-spam fee payable in :
- GAS - Native BTNS token
- XCP - Counterparty
- BTC - Bitcoin

Anti-spam payment mechanisms :
- Fee is donated to X / Y / Z causes
- Fee Destroyed (GAS/XCP=Destroy, BTC=Burn)
- Fee Goes to miner (BTC Only at first)

Counterparty `custom_outputs` issue
https://github.com/CounterpartyXCP/counterparty-lib/issues/1214

Freewallet `checkDonate()` code
https://github.com/jdogresorg/freewallet-desktop/blob/master/js/freewallet-desktop.js#L6452-L6547

**Comment by ghost** [comment]
for example a token Named BTNS
Can we try something new :
for example :
#1 A command DEPLOY including  TWO ACTIONs (DEPLOY+100%Mint OUT) which only need one time broadcast
#2 A command  MINT including TWO ACTIONS (CLAIM+ AIRDROP) fees paid depends on QTY claimed, fees owned by BTNS/BTC contract mostly 

 MINT  cost 4M USD totally, 90% given to provide Liquidity via X*y=K swap like uniswap 
another 10% given to developer as we want to see  infrastructure builder  being more motivated 

anyway try to save fees for program don't just burn fees to miner

regarding payment way i strongly try to  adopt what unisat are using 

pay with Bitcoin
pay with freewallet (LTN)
Pay with Mixpay(Binance Pay)

![WeChatdb2ad5c8a969dd8943debfea8f6708b1](https://github.com/jdogresorg/Broadcast-Token-Naming-System/assets/107408080/c51681e3-e456-49a7-a99e-adc40f589166)

**Comment by jdogresorg** [comment]
thanks for your feedback, but this was more of a note to myself....

BTNS does not have any DEPLOY... it has ISSUE, and yes, you can mint supply at the time of issuance.

Please wait until BTNS indexer and tools comes out to give feedback.

**Comment by Soulchild07** [comment]
thx

**Reply by mymrUw-pavviz-febvi6** [reply]
Obaretin

**Comment by jdogresorg** [comment]
Additional relevant thoughts on providing address specific options that would allow for functionality discussed in issue #34 .
